src/town_map.h

```diff
@@ -74,6 +74,28 @@ inline void SetHouseType(Tile t, HouseID house_id)
 	SB(t.m8(), 0, 12, house_id);
 }
 
+/**
+ * Check if the house is protected from removal by towns and AIs.
+ * @param t The tile.
+ * @return If the house is protected.
+ */
+inline bool IsHouseProtected(Tile t)
+{
+	assert(IsTileType(t, MP_HOUSE));
+	return HasBit(t.m3(), 5);
+}
+
+/**
+ * Set a house as protected from removal by towns and AIs.
+ * @param t The tile.
+ * @param protect Should the house be protected?
+ */
+inline void SetHouseProtected(Tile t, bool protect)
+{
+	assert(IsTileType(t, MP_HOUSE));
+	SB(t.m3(), 5, 1, !!protect);
```

Comment:
I do not like `!!`, but I guess you copied it from somewhere? I am a fun of `protect ? 1 : 0`. But that is just me, possibly.